Build failure: simavr (aarch64)

```shell
nix shell github:nixos/nixpkgs#simavr
```
... ends up failing on gcc12, saying:
```
error: builder for '/nix/store/cjsj26967i48f5qws2cd33421lwzg7s9-avr-stage-static-gcc-12.2.0.drv' failed with exit code 2;
       last 10 log lines:
       > checking whether the thread-local storage support is from emutls... yes
       > checking for init priority support... no
       > configure: updating cache ./config.cache
       > configure: creating ./config.status
       > config.status: creating Makefile
       > config.status: creating auto-target.h
       > config.status: executing default commands
       > Adding multilib support to Makefile in ../../../../../gcc-12.2.0/libgcc
       > with_multisubdir=avrtiny/long-double32
       > make: *** [Makefile:1015: all] Error 2
       For full logs, run 'nix log /nix/store/cjsj26967i48f5qws2cd33421lwzg7s9-avr-stage-static-gcc-12.2.0.drv'.
error: 1 dependencies of derivation '/nix/store/gf2416s4r1yanhpiyq3iqh0ppxylhj9n-avr-stage-static-gcc-wrapper-12.2.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ivch3v6md9v53qn6zvxv4j09n5nggh43-stdenv-darwin.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gpjjgb9a4da1yadlbn2ffqla9mmn5yr1-avr-libc-avr-2.1.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jjgfx3a1aw4s07yhklm2dsfyqv0fc9zi-simavr-1.7.drv' failed to build
```
... with the actual reason being:
```
clang-11: error: unknown argument: '-nodefaultrpaths'
```

This seems somehow related to MacOS 13:
- https://github.com/iains/gcc-12-branch/issues/6

--- a/pkgs/development/compilers/gcc/12/default.nix
+++ b/pkgs/development/compilers/gcc/12/default.nix
@@ -64,7 +64,10 @@ let majorVersion = "12";
         ../gnat-cflags-11.patch
         ../gcc-12-gfortran-driving.patch
         ../ppc-musl.patch
-      ] ++ optional (stdenv.isDarwin && stdenv.isAarch64) (fetchpatch {
+      ]
+      # We only apply this patch when building a native toolchain for aarch64-darwin, as it breaks building
+      # a foreign one: https://github.com/iains/gcc-12-branch/issues/18
+      ++ optional (stdenv.isDarwin && stdenv.isAarch64 && buildPlatform == hostPlatform && hostPlatform == targetPlatform) (fetchpatch {
         name = "gcc-12-darwin-aarch64-support.patch";
         url = "https://github.com/Homebrew/formula-patches/raw/1d184289/gcc/gcc-12.2.0-arm.diff";
         sha256 = "sha256-omclLslGi/2yCV4pNBMaIpPDMW3tcz/RXdupbNbeOHA=";

